<!DOCTYPE html>
<html>
<head>
    <title></title>
</head>
<body>
<script src="../common/JSXTransformer.js"></script>
<script src="../common/react16.js"></script>
<script src="../common/react-dom16.js"></script>
<script type="text/jsx">
    /**
        componentDidCatch会捕获子组件的错误
        从而可以给定一个回退页面
    */
    const {Component} = React;
    class Test extends Component{
        constructor(props){
            super(props);
            this.state={
                show:'show'
            }
        }
        componentDidCatch(){
            this.setState({
                'show':'error'
            })
        }
        render(){
            const context=this.state.show=='error'?this.state.show:this.props.children;
            return (
                <div>
                    {context}
                </div>
            )
        }
    }
    const T=props=>(
        <div>{props.name.age}</div>
    )
    ReactDOM.render(<Test><T /></Test>,document.body);
</script>
</body>
</html>